src/pfe/file-watcher/server/test/test.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+# Colors for success and error messages
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+BLUE='\033[0;36m'
+RESET='\033[0m'
+
+function usage {
+    me=$(basename $0)
+    cat <<EOF
+Usage: $me: [-<option letter> <options value> | -h]
+Options:
+    -t # Test type, currently support 'local' and 'kube' - Mandatory
+    -b # Test branch - Mandatory
+    -s # Test suite - Mandatory
+    -i # Ignore post cleanup, ignore post test automation cleanup for debug purpose, currently support 'y' and 'n' - Optional
+    -c # Test configuration - currently support 'setup' for install and uninstall PFE container - Optional
+    -h # Display the man page
+EOF
+}
+
+while getopts "t:b:s:i:c:h" OPTION; do
+    case "$OPTION" in
+        t) 
+            TEST_TYPE=$OPTARG
+            # Check if test type argument is corrent
+            if [[ ($TEST_TYPE != "local") && ($TEST_TYPE != "kube") ]]; then
+                echo -e "${RED}Test type argument is not correct. ${RESET}\n"
+                usage
+                exit 1
+            fi
+            ;;
+        b) 
+            TEST_BRANCH=$OPTARG
+            ;;
+        s) 
+            TEST_SUITE=$OPTARG
+            ;;
+        i)
+            IGNORE_POST_CLEANUP=$OPTARG
+            # Check if ignore post cleanup argument is corrent
+            if [[ ($IGNORE_POST_CLEANUP != "y") && ($IGNORE_POST_CLEANUP != "n") ]]; then
+                echo -e "${RED}Ignore post cleanup argument is not correct. ${RESET}\n"
+                usage
+                exit 1
+            fi
+            ;;
+        c) 
+            TEST_CONFIGURATION=$OPTARG
+            # Check if test configuration argument is corrent
+            if [[ ($TEST_CONFIGURATION != "setup") ]]; then
+                echo -e "${RED}Test configuration argument is not correct. ${RESET}\n"
+                usage
+                exit 1
+            fi
+            ;;
+        *)
+            usage
+            exit 0
+            ;;
+    esac
+done
+
+# Chekc mandatory arguments have been set up
+if [[ (-z $TEST_TYPE) || (-z $TEST_BRANCH) || (-z $TEST_SUITE) ]]; then
+    echo -e "${RED}Mandatory arguments are not set up. ${RESET}\n"
+    usage
+    exit 1
+fi
+
+if [[ (-z $TEST_CONFIGURATION) ]]; then
+    # Run test cases
+    ./scripts/exec.sh -t $TEST_TYPE -s $TEST_SUITE
```

Comment:
It'll be nice if you could put all these exec, setup and other cmds in this script into functions so that look like setup.sh

Reply:
I can do the change